File: chapitre-05/exemple-zombie-1.c
// ------------------------------------------------------------------
// exemple-zombie-1.c
// Fichier d'exemple du livre "Developpement Systeme sous Linux"
// (C) 2000-2019 - Christophe BLAESS <christophe@blaess.fr>
// https://www.blaess.fr/christophe/
// ------------------------------------------------------------------

#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

int main (void)
{
	int   i;
	pid_t pid;
	char  command[128];

	if ((pid = fork()) < 0) {
		fprintf(stderr, "error during fork()\n");
		exit(EXIT_FAILURE);
	}

	if (pid == 0) {
		/* child process */
		sleep(2);
		fprintf(stdout, "Child process %ld terminates\n", (long) getpid());
		exit(EXIT_SUCCESS);

	} else {
		/* parent process */
		snprintf(command, 128, "ps %ld", (long)pid);
		for (i = 0; i < 5; i ++) {
			system(command);
			sleep(1);
		}
	}
	return EXIT_SUCCESS;
}

File: chapitre-05/exemple-zombie-1.py
#! /usr/bin/python
# ------------------------------------------------------------------
# exemple-zombie-1.py
# Fichier d'exemple du livre "Developpement Systeme sous Linux"
# (C) 2015-2019 - Christophe BLAESS <christophe@blaess.fr>
# https://www.blaess.fr/christophe/
# ------------------------------------------------------------------

from __future__ import print_function
import os
import sys
import time

pid = os.fork()

if (pid == 0):
	time.sleep(2)
	print("Child process will terminate now!")

else:
	for i in range(1,5):
		os.system("ps {0}".format(pid))
		time.sleep(1)

sys.exit(0)
